src/docfx/SubCommands/BuildCommand.cs

```diff
@@ -253,10 +253,7 @@ private static void MergeOptionsToConfig(BuildCommandOptions options, BuildJsonC
         private static void MergeGitContributeToConfig(BuildJsonConfig config)
         {
             GitDetail repoInfoFromBaseDirectory = GitUtility.GetGitDetail(Path.Combine(Environment.CurrentDirectory, config.BaseDirectory));
-            if (repoInfoFromBaseDirectory?.LocalWorkingDirectory != null)
-            {
-                config.GlobalMetadata["_baseDirectory"] = config.BaseDirectory;
-            }
+            EnvironmentContext.BaseDirectory = config.BaseDirectory;
```

Comment:
this method seems to be specific for merging git contributors, so move this line outside.